### PR TITLE
Minor changes to the safehtml post.

### PR DESCRIPTION
**Pull request by kele**
All optional, of course.

I've reordered the sections in order of importance.

### CODE REVIEW

File: content/post/go-safehtml.md [hunk 1]
```diff
@@ -37,24 +37,18 @@ This is the root package and it is just providing types that are [safe by constr
...
 
-This guarantees that every single instance of the HTML type is known to be safe, because there is no unsafe way to construct it. The [Script type](https://godoc.org/github.com/google/safehtml#Script) behaves similarly but instead of having templates it can only be built from constants or data. To express the concept of "compile time constant" it has [constructors that take unexported string types](https://godoc.org/github.com/google/safehtml#ScriptFromConstant), so the only way to call them is with a [string literal](https://golang.org/ref/spec#String_literals) (I find this to be a very neat trick).
+This guarantees that every single instance of the HTML type is known to be safe. The [Script type](https://godoc.org/github.com/google/safehtml#Script) behaves similarly but instead of having templates it can only be built from constants or data. To express the concept of a "compile time constant" it has [constructors that take unexported string types](https://godoc.org/github.com/google/safehtml#ScriptFromConstant), so the only way to call them is with a [string literal](https://golang.org/ref/spec#String_literals) (I find this to be a very neat trick).
```
Comment (kele) [inline content/post/go-safehtml.md:40]: I added "a" before "compile time constant" here. This is the only change.

File: content/post/go-safehtml.md [hunk 2]
```diff
@@ -37,24 +37,18 @@ This is the root package and it is just providing types that are [safe by constr
...
 
 All other types in this package follow a similar pattern.
 
-### safehtml/legacyconversions
-This package should only be used to transition to the safe API. It blesses any arbitrary string to be a safe type so that transition to safehtml can be very quick and all new code will be safe. <ins>**Once the migration has happened the use of this package should be prevented**</ins>. As the name states: this is just for legacy code, no new code should be using it, and **all usages of this packages should gradually be refactored to use safe constructors instead**.
```
Comment (kele) [inline content/post/go-safehtml.md:45]: Reordered, no other changes.

File: content/post/go-safehtml.md [hunk 3]
```diff
@@ -37,24 +37,18 @@ This is the root package and it is just providing types that are [safe by constr
...
-
-### safehtml/raw
```
Comment (kele) [inline content/post/go-safehtml.md:47]: Reordered, no other changes.

File: content/post/go-safehtml.md [hunk 4]
```diff
@@ -37,24 +37,18 @@ This is the root package and it is just providing types that are [safe by constr
...
-Importing this package should be prevented. Anything outside of the "safehtml/" directory tree should not have visibility of this package.
-
-### safehtml/safehtmlutil
```
Comment (kele) [inline content/post/go-safehtml.md:50]: Reordered, no other changes.

File: content/post/go-safehtml.md [hunk 5]
```diff
@@ -37,24 +37,18 @@ This is the root package and it is just providing types that are [safe by constr
...
-Yes, I know, not a good name. Consider that this package, like the previous one, should also not be imported outside of safehtml and it was just created to reduce code duplication and avoid cyclic dependencies. I agree this could have been named or structured differently, but since you're never going to interact with this package it should not bother you too much.
-
 ### safehtml/template
-This is your real "html/template" replacement and it is the package everyone should be using. If "legacyconversions" and "uncheckedconversions" are not used and **all your HTML responses are generated by this package** you have the guarantee there is not going to be any server-side XSS in your products.
+This is your real "html/template" replacement and it is the package everyone should be using. If "legacyconversions" and "uncheckedconversions" (read below) are not used and **all your HTML responses are generated by this package** you have the guarantee there is not going to be any server-side XSS in your products.
```
Comment (kele) [inline content/post/go-safehtml.md:45]: Added (read below) because I reordered the packages.

File: content/post/go-safehtml.md [hunk 6]
```diff
@@ -63,10 +57,22 @@ This is the most nuanced matter. Sometimes the safehtml API is too inconvenient
 
 For these **very rare** situations you can use this package. Importing it should be restricted to a hand-picked set of dependants and every new import should require some security-aware folks to review it.
 
-Usages of this package are your single point of failure, so make sure you only use it when necessary. (This sentence assumes you will eventually get rid of legacy conversions)
+Make sure that the usage **is safe and will stay safe**, as uncheckedconversions do not increase safety. They are just there to inform the compiler that you have reviewed the code and want it to be trusted. Follow these guidelines:
```
Comment (kele) [inline content/post/go-safehtml.md:60]: Added this section based mostly on go/hardening-comments guidance I've written recently.

**Review by empijei — APPROVED** [review]

**Pull request merged**